Assets/Scripts/Entity/Enemy/MovementBehaviors/WanderMovement.cs

```diff
@@ -21,43 +21,67 @@ private void Awake()
     }
 
 
+    /* Generates a random direction in which the agent should head.
+     */  
+    Vector3 GenerateDestination()
+    {
+        // Pick a random direction for the agent to head in
+        Vector3 randomDirection = Random.insideUnitSphere * wanderRadius;
+        randomDirection += transform.position;
+
+        // Sample points on the navmesh until we get a hit (this usually terminates in one iteration except for an edge case)
+        NavMeshHit navMeshHit;
+        do
```

Comment:
**Change requested here**

Also, some times a valid position can not be obtained (I assume when the enemy is right up against a wall or obstacle so that the `wanderRadius` fits inside the obstacle or outside the wall), which results in this spin locking the game, effectively freezing it permanently. 

The root problem is how the random point on the navmesh is being determined. If `Random.insideUnitSphere` just so happens to be the extreme, then that multiplied by `wanderRadius` will put that point exactly `wanderRadius` away. This means that when `SamplePosition` is called, if the NavMeshAgent just so happens to be at a wall or obstacle that is large enough, and the `randomDirection` is away from the navmesh, then `SamplePosition` will not find a point on the navmesh.

One quick and dirty solution is to just decrease the `wanderRadius` when it is multiplied into `randomDirection` or increase the radius given to the `SamplePosition` method. This way, the point on which the NavMeshAgent currently resides on will always be within the query radius for `SamplePosition` and thus will be returned instead of erroring out.

Though, Unity's NavMeshAgent seems to handle the error by itself by just ignoring that you passed it an invalid destination. So just leaving it alone should be fine (Probably with a comment though).

But the loop does have to go, because that is actually crashing the game.

Reply:
Awesome, thanks for the feedback! Yeah, I'll get rid of this loop. I figured it was the problem because I introduced it as a "solution" to the `CalculatePolygonPath: invalid target position { Infinity, Infinity, Infinity }.` error I was getting in Unity. But it seems it handles that on its own, so we're good :)